packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -21,7 +21,30 @@ import 'template.dart';
 ///
 /// This defines interfaces common to iOS and macOS projects.
 abstract class XcodeBasedProject extends FlutterProjectPlatform  {
-  static const String _hostAppProjectName = 'Runner';
+  static const String defaultHostAppProjectName = 'Runner';
+  String _hostAppProjectName = defaultHostAppProjectName;
+
+  String get hostAppProjectName {
+    if (_hostAppProjectName != 'Runner') {
+      return _hostAppProjectName;
+    }
+    List<FileSystemEntity> contents;
+    try {
+      contents = hostAppRoot.listSync();
+      for (final Directory entity in contents.whereType<Directory>()) {
+        // On certain volume types, there is sometimes a stray `._Runner.xcworkspace` file.
+        // Find the first non-hidden xcworkspace and return the directory.
+        if (globals.fs.path.extension(entity.path) == '.xcworkspace' &&
+            !globals.fs.path.basename(entity.path).startsWith('.')) {
+          _hostAppProjectName = globals.fs.path.basenameWithoutExtension(entity.path);
+          return _hostAppProjectName!;
+        }
+      }
+    } catch (e) {
```

Comment:
i removed the try catch - originally i added it because the cocoapod tests was throwing an exception because it tried to grab the hostAppProjectName with an empty test directory. I just added the corresponding macos and ios directories to the test directory, so it should be fine now.